libs/facilities/backend/insight-hub/src/lib/services/iot-time-series.service.ts

```diff
@@ -0,0 +1,45 @@
+import { HttpService } from '@nestjs/axios';
+import { Inject, Injectable } from '@nestjs/common';
+import { IInsightHub } from 'common-backend-models';
+import { defaults } from 'lodash';
+import { catchError, map, Observable } from 'rxjs';
+
+import { ITimeSeriesRequestParameter } from '../models/interfaces/time-series-request.interface';
+import { INSIGHT_HUB_OPTIONS } from '../tokens';
+
+/**
+ * Service to interact with the IoT Time Series API.
+ */
+@Injectable()
+export class IotTimeSeriesService {
+	constructor(
+		private readonly _httpClient: HttpService,
+		@Inject(INSIGHT_HUB_OPTIONS)
+		private readonly _insightHubOptions: IInsightHub,
+	) {}
+
+	/**
+	 * Allows to get the time series data from the IoT Time Series API.
+	 */
+	public getTimeSeriesData<SelectType, ReturnType>(
```

Comment:
please also add more type safety

I guess it should look like that

```suggestion
	public getTimeSeriesData<
             SelectType extends Record<PropertyKey, any>,
             ReturnType extends { time: Date; [key: PropertyKey]: any }
          >(
```

Reply:
Good suggestion, however I ll export the return type then into its own interface